components/dashboard/src/projects/ConfigureProject.tsx

```diff
@@ -179,9 +179,7 @@ export default function () {
         <div className="h-20 px-6 bg-gray-50 dark:bg-gray-800 border-t border-gray-200 dark:border-gray-600 flex space-x-2">
           {prebuildWasTriggered && <PrebuildInstanceStatus prebuildInstance={prebuildInstance} isDark={isDark} />}
           <div className="flex-grow" />
-          {(prebuildInstance?.status.phase === "stopped" && !prebuildInstance?.status.conditions.failed)
-              ? <a className="my-auto" href={`/#${project?.cloneUrl}`}><button className="secondary">New Workspace</button></a>
-              : <button disabled={true} className="secondary">New Workspace</button>}
+          <a className="my-auto" href={`/#${project?.cloneUrl}`}><button className="secondary">New Workspace</button></a>
```

Comment:
Instead of always enabling `New Workspace`, I propose to only enable it when:

- There was a Git-based config (new with this PR, requested by https://github.com/gitpod-io/gitpod/issues/5591)
- User ran a Prebuild, which had no errors, and stopped successfully (previous code)

I think we don't want to enable the `New Workspace` button when:

- User ran a Prebuild, but it's not quite finished yet (so starting a workspace wouldn't have the config)
- User ran a Prebuild, but it had an error (starting a workspace would not use any prebuild)
- User didn't run a Prebuild yet, and config was auto-detected (what you see in the editor isn't saved yet, and starting a new workspace would give you an unconfigured workspace)

```suggestion
                    {((!isDetecting && isEditorDisabled) || (prebuildInstance?.status.phase === "stopped" && !prebuildInstance?.status.conditions.failed))
              ? <a className="my-auto" href={`/#${project?.cloneUrl}`}><button className="secondary">New Workspace</button></a>
              : <button disabled={true} className="secondary">New Workspace</button>}
```

Reply:
Great catch and proposal @jankeromnes!  💯 

All points are valid. Have also verified most of the issues you mentioned in https://github.com/gitpod-io/gitpod/pull/5654#issuecomment-921665768.

Reply:
Now implemented in https://github.com/gitpod-io/gitpod/pull/5907